src/urllib3/_async/connectionpool.py

```diff
@@ -383,7 +383,13 @@ def _raise_timeout(self, err, url, timeout_value):
         host = self.host
         port = self.port
         scheme = self.scheme
-        request.add_host(host, port, scheme)
+
+        # When given a URL wit a trailing dot for the hostname part:
+        # "https://example.com./", urllib3 will strip off the dot and use the name
+        # without a dot internally and send it dot-less in HTTP Host: headers and in
+        # the TLS SNI field.
```

Comment:
I realize that you've used the comment from https://github.com/curl/curl/commit/3a6142865fec3c54cd7081ada86c93c135e4b32f but it's not perfect here. First, there's a typo where "wit" should be "with". (This [has been fixed in curl](https://github.com/curl/curl/blob/938c17d07a8c9aae52bf58ff9713f4197d2537cb/docs/KNOWN_BUGS#L180).)

Second, I don't think you need to mention SNI, it's already handled in `_wrap_socket`:

https://github.com/python-trio/urllib3/blob/652e0fb5d4c6871690de91e83863a4a0679597a2/src/urllib3/_async/connection.py#L364-L368

See https://github.com/urllib3/urllib3/issues/1254 and https://github.com/curl/curl/issues/716 to understand why we're doing this. So I think we can use something like:

```
# Stripping trailing dots from the Host header to keep the HTTP Host in sync
# between SNI and HTTP to avoid confuse servers
# https://github.com/urllib3/urllib3/issues/1254
```

@njsmith Thoughts?

Reply:
I have updated the comment.

Reply:
Sorry I meant "to avoid confusing servers"!